mappings/net/minecraft/client/search/SearchManager.mapping

```diff
@@ -2,14 +2,22 @@ CLASS net/minecraft/unmapped/C_zqzigdge net/minecraft/client/search/SearchManage
 	FIELD f_exfajduf RECIPE_OUTPUT Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;
 	FIELD f_hpwjudey ITEM_TOOLTIP Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;
 	FIELD f_jivcozuv ITEM_TAG Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;
-	FIELD f_xowozfgq instances Ljava/util/Map;
-	METHOD m_jmupjdnd (Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;)Lnet/minecraft/unmapped/C_zqzigdge$C_lpzuavsm;
+	FIELD f_xowozfgq searcherProviders Ljava/util/Map;
+	METHOD m_jmupjdnd getProvider (Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;)Lnet/minecraft/unmapped/C_zqzigdge$C_lpzuavsm;
 		ARG 1 key
-	METHOD m_jrasqirm put (Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;Lnet/minecraft/unmapped/C_zqzigdge$C_faknnetb;)V
+	METHOD m_jrasqirm registerProvider (Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;Lnet/minecraft/unmapped/C_zqzigdge$C_faknnetb;)V
 		ARG 1 key
+		ARG 2 provider
 	METHOD m_xgrlmmoi get (Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;)Lnet/minecraft/unmapped/C_inuoeylq;
 		ARG 1 key
 	METHOD m_yhcpnstp refreshSearch (Lnet/minecraft/unmapped/C_zqzigdge$C_gvofkqkn;Ljava/util/List;)V
 		ARG 1 key
 		ARG 2 searchables
+	CLASS C_faknnetb RefreshingSearcherProviderFunction
```

Comment:
```java
@Environment(EnvType.CLIENT)
public interface RefreshingSearcherProviderFunction<T> extends Function<List<T>, RefreshingSearcher<T> {
}
```

Reply:
this PR looks fantastic, thank you! my only comment: what's stopping you from calling this `Factory` instead of `ProviderFunction`? I think that would be better for conciseness